Mathlib/CategoryTheory/Sites/Equivalence.lean

```diff
@@ -34,16 +34,62 @@ sufficiently small limits in the sheaf category on the essentially small site.
 
 -/
 
-universe u
+universe w v u
 
 namespace CategoryTheory
 
-open Functor Limits GrothendieckTopology
+open Functor Limits GrothendieckTopology Presheaf
 
 variable {C : Type*} [Category C] (J : GrothendieckTopology C)
 variable {D : Type*} [Category D] (K : GrothendieckTopology D) (e : C ≌ D)
 variable (A : Type*) [Category A]
 
+namespace Presheaf
+
+variable (H : C ⥤ D) [ConcreteCategory A] {F G : Dᵒᵖ ⥤ A} (f : F ⟶ G)
+
+lemma isLocallyInjective_whisker_isCocontinuous [H.IsCocontinuous J K] [IsLocallyInjective K f] :
+    IsLocallyInjective J (whiskerLeft H.op f) where
+  equalizerSieve_mem x y h := H.cover_lift J K (equalizerSieve_mem K f x y h)
+
+lemma isLocallyInjective_of_whisker_coverPreserving_isCoverDense (hH : CoverPreserving J K H)
+    [H.IsCoverDense K] [IsLocallyInjective J (whiskerLeft H.op f)] : IsLocallyInjective K f where
+  equalizerSieve_mem {X} a b h := by
+    apply K.transitive (Functor.IsCoverDense.is_cover (G := H) (K := K) X.unop)
```

Comment:
Please use `H.is_cover_of_isCoverDense`. (In general, we should almost never use `TypeClass.field`, but API lemmas with suitable explicit variables and `[TypeClass]` as a parameter.)